### DISCUSSION TITLE
Updates overwrite excluded apps file

### DISCUSSION THREAD

**Discussion by MBstatsara** [Romanitho/Winget-AutoUpdate#650] [General]
Hello
It seems as the wingetauto updater app patches to the latest version, the excluded apps file is being overwritten.
This is causing the excluded apps we have centrally to be older than the file it is currently using, so it doesnt use it. (its just used the one from the patch)
This means we are having to update our excluded apps file with no changes to make it always being used.

How do we stop the excluded apps file being overwritten when a patch occurs?

**Comment by KnifMelti** [comment]
The issue was not present in today's pre-release version 1.20.2-1 for me, nor has it occurred for us previously with stable version releases.
Within the `Update-WAU.ps1` script, the `WAU.zip` that is downloaded includes the file `excluded_apps.txt`. This file is situated outside the `$WorkingDir\WAU_update\Winget-Autoupdate` folder, which is the directory from where the new files are sourced.

**Reply by KnifMelti** [reply]
@MBstatsara answered:
Sorry have moved this to the autoupdate rep, as didnt realise i was in the installer section.
Was fine on all other builds but can confirm this latest patch last night
WAU 1.20.2-1 is the build we moved to..

Previously the excluded apps file hasnt been overwritten, but i have about 50 PCs where it did today, i had to edit the central repo again to get the file changed back

**Reply by KnifMelti** [reply]
You shouldn't use pre-release for production...
...and I didn't have this issue with my local file and central repo (https).
@Romanitho ?

**Reply by KnifMelti** [reply]
@MBstatsara answered:
There is no option to disable pre-release?

**Reply by KnifMelti** [reply]
If you started by installing a pre-release on clients, they will update to every pre-release rolling out.
Installing a stable version updates to stable versions.
The setting is in `HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate\WAU_UpdatePrerelease` (`1 or 0`) and that can be managed by **GPO**

**Reply by MBstatsara** [reply]
Thanks, will try adding that in now.

**Reply by Romanitho** [reply]
100% agree. Don't use prerelease in prod. We are testing new features, bug troubleshoot, and other stuffs...

**Reply by MBstatsara** [reply]
but as far as we are aware, the files in place shouldnt be over written, i also noted the systems app file was gone as well, is the noclean option something we need to do, or is that just for the install and not update

**Reply by Romanitho** [reply]
And yes, by default if you install a prerelease version, you enable prelease update channel.

**Reply by KnifMelti** [reply]
> but as far as we are aware, the files in place shouldnt be over written, i also noted the systems app file was gone as well, is the noclean option something we need to do, or is that just for the install and not update

No, the **included/excluded_apps.txt** should not be overwritten.
I don't follow.. ..what is `the systems app file`? [EDIT] Ah it's **winget_system_apps.txt** exported for comparing the user installed apps...
The `-NoClean` parameter is just for installing/uninstalling.

**Reply by KnifMelti** [reply]
I tested several upgrades in my virtual environment but was unable to replicate the issue @MBstatsara encountered.

**Reply by MBstatsara** [reply]
Thanks - lets close this for now, i think it might be a issue we need to pickup if it happens again, but i cant see why that was using the default excluded apps file..
